crypto/store/store_register.c

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright 1995-2016 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <string.h>
+#include <ctype.h>
+
+#include <openssl/err.h>
+#include <openssl/lhash.h>
+#include "store_local.h"
+
+/******************************************************************************
+ *
+ *  Functions for manipulating STORE_LOADERs
+ *
+ *****/
+
+STORE_LOADER *STORE_LOADER_new(ENGINE *e)
+{
+    STORE_LOADER *res = OPENSSL_zalloc(sizeof(*res));
+
+    if (res == NULL)
+        STOREerr(STORE_F_STORE_LOADER_NEW, ERR_R_MALLOC_FAILURE);
+
+    res->engine = e;
+    return res;
+}
+
+const ENGINE *STORE_LOADER_get0_engine(const STORE_LOADER *loader)
+{
+    return loader->engine;
+}
+
+int STORE_LOADER_set0_scheme(STORE_LOADER *loader, const char *scheme)
```

Comment:
Errr, I though that `set1` implied that and that `set0` implies an expectation that the caller keep that memory area alive and untouched for the life time of the loader.